Content.Shared/Teleportation/Systems/SharedPortalSystem.cs

```diff
@@ -157,6 +142,11 @@ private void OnEndCollide(EntityUid uid, PortalComponent component, ref EndColli
 
         Transform(subject).Coordinates = target;
 
+        // if this portal isn't parented to the grid, do so to prevent funny stuff
+        var grid = target.GetGridUid(EntityManager);
+        if (grid != null)
+            _transform.SetParent(subject, grid.Value);
```

Comment:
This looks hacky and shouldn't be here.

Reply:
it is absolutely necesaary
if you go out of portal then close it, you get deleted if you dont buckle yourself to something

Reply:
If you have the same parent as the portal then deleting the portal shouldn't delete you unless there's some other bug going on, in which case this would still be a bandaid.

Reply:
shouldnt deleting any entity by design delete its children?

Reply:
Yeah?

So what happens atm:
Subject coordinates set to target coordinates (expensive).
If target on a grid now set the subject's coordinates to grid (expensive).

If you just want to attach it to the grid then do that directly.